### PR TITLE
Comment controller

### PR DESCRIPTION
**Pull request by NeuralAlchemist**
Implemented comment controller with documentation.

### CODE REVIEW

File: src/main/java/se/kth/sda/skeleton/comments/Comment.java [hunk 1]
```diff
@@ -12,11 +18,42 @@
...
     private Long id;
 
+    @Column(nullable = false, length=1500)
+    @NotBlank
+    private String contentText;
+
+    @ManyToOne
+    @JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
+    @JsonIdentityReference(alwaysAsId = true)
+    @JoinColumn(nullable = false)
```
Comment (eeels22) [inline src/main/java/se/kth/sda/skeleton/comments/Comment.java:27]: also add  @NotNull annotation

File: src/main/java/se/kth/sda/skeleton/comments/CommentController.java [hunk 2]
```diff
@@ -1,8 +1,72 @@
...
+        commentService.createComment(id, comment);
+        return ResponseEntity.status(HttpStatus.CREATED).body(comment);
+    }
+
+    /**
+     * Handler method for endpoint "/comments/{id}" with GET HttpRequest. Returns a {@link ResponseEntity} containing a Comment with the
+     * given {@code id} and HTTP status {@code OK}.
+     * @param id the id used to find the Comment associated with it
+     * @return {@link ResponseEntity} containing a Comment with the given {@code id} and HTTP status {@code OK}
+     */
```
Comment (eeels22) [inline src/main/java/se/kth/sda/skeleton/comments/CommentController.java]: javadoc does not match the method. Doc relates to a GET comment by id method but the code is for a PUT method

File: src/main/java/se/kth/sda/skeleton/comments/CommentRepository.java [hunk 3]
```diff
@@ -2,10 +2,10 @@
 
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.stereotype.Repository;
-
```
Comment (eeels22) [inline src/main/java/se/kth/sda/skeleton/comments/CommentRepository.java]: I think we can revert the changes on this page as the spacing before is more standard.

File: src/main/java/se/kth/sda/skeleton/comments/CommentService.java [hunk 4]
```diff
@@ -1,7 +1,75 @@
 package se.kth.sda.skeleton.comments;
 
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
-
+import se.kth.sda.skeleton.exception.ResourceNotFoundException;
+import se.kth.sda.skeleton.posts.Post;
+import se.kth.sda.skeleton.posts.PostRepository;
+import java.util.List;
+/**
```
Comment (eeels22) [inline src/main/java/se/kth/sda/skeleton/comments/CommentService.java:10]: add empty line above javadoc

File: src/main/java/se/kth/sda/skeleton/comments/CommentService.java [hunk 5]
```diff
@@ -1,7 +1,75 @@
...
+ * Represents the service layer. It contains defined functionalities according to business logic for Comments.
+ */
 @Service
 public class CommentService {
+    private CommentRepository commentRepository;
+    private PostRepository postRepository;
+    /**
```
Comment (eeels22) [inline src/main/java/se/kth/sda/skeleton/comments/CommentService.java:18]: add empty line above javadoc

File: src/main/java/se/kth/sda/skeleton/comments/CommentService.java [hunk 6]
```diff
@@ -1,7 +1,75 @@
...
+        commentRepository.delete(comment);
+    }
+
+    /**
+     * Creates a Comment for a Post with the given {@code postId} or throws a {@link ResourceNotFoundException} if there is no Post with the given {@code postId}.
+     * @param postId the Post for whom a new Comment is created
+     * @param comment the Comment that will be associated with the Post whose id is {@code postId}
+     * @throws ResourceNotFoundException if there is no Post with the given {@code postId}
+     */
+    public void createComment(Long postId, Comment comment){
```
Comment (janko101) [inline src/main/java/se/kth/sda/skeleton/comments/CommentService.java]: Should the method return a Comment object?

File: src/main/java/se/kth/sda/skeleton/comments/Comment.java [hunk 7]
```diff
@@ -1,9 +1,15 @@
...
+import com.fasterxml.jackson.annotation.*;
+import javax.persistence.*;
+import javax.validation.constraints.*;
+import se.kth.sda.skeleton.posts.Post;
+
+/**
+ * Represents a comment on a {@link Post} as a JPA Entity. This implementation of comment will autogenerate a primary key of type
+ * {@link Long} to indicate the current Comment entity. A Comment has a non-null field called {@code relatedPost}. The {@code relatedPost}
+ * is a {@link Post} to which the Comment is related. The body of a {@code Comment} cannot be null and is limited to 1500 characters.
+ */
```
Comment (eeels22) [inline src/main/java/se/kth/sda/skeleton/comments/Comment.java]: remove empty line between end of javadoc and start of class declaration

File: src/main/java/se/kth/sda/skeleton/comments/CommentController.java (hunk 2 above)
Comment (eeels22) [reply src/main/java/se/kth/sda/skeleton/comments/CommentController.java]: Add a Get comment by id method

**Review by eeels22 — CHANGES_REQUESTED** [review]
Thanks for these update. Made some comments.

**Review by janko101 — APPROVED** [review]

File: src/main/java/se/kth/sda/skeleton/comments/Comment.java (hunk 1 above)
Comment (NeuralAlchemist) [reply src/main/java/se/kth/sda/skeleton/comments/Comment.java:28]: done

Comment (NeuralAlchemist) [reply src/main/java/se/kth/sda/skeleton/comments/Comment.java:27]: added

File: src/main/java/se/kth/sda/skeleton/comments/CommentService.java (hunk 4 above)
Comment (NeuralAlchemist) [reply src/main/java/se/kth/sda/skeleton/comments/CommentService.java:10]: done

File: src/main/java/se/kth/sda/skeleton/comments/CommentService.java (hunk 5 above)
Comment (NeuralAlchemist) [reply src/main/java/se/kth/sda/skeleton/comments/CommentService.java:18]: yup

File: src/main/java/se/kth/sda/skeleton/comments/CommentService.java (hunk 6 above)
Comment (NeuralAlchemist) [reply src/main/java/se/kth/sda/skeleton/comments/CommentService.java]: changed it so it returns the saved comment, updated in comment controller as well.

File: src/main/java/se/kth/sda/skeleton/comments/Comment.java (hunk 7 above)
Comment (NeuralAlchemist) [reply src/main/java/se/kth/sda/skeleton/comments/Comment.java]: done

File: src/main/java/se/kth/sda/skeleton/comments/CommentController.java (hunk 2 above)
Comment (NeuralAlchemist) [reply src/main/java/se/kth/sda/skeleton/comments/CommentController.java]: done

**Review by eeels22 — APPROVED** [review]
Thanks for the changes!
